### PR TITLE
Dispose of old event listeners correctly

### PR DESCRIPTION
**Pull request by owennw**
copy of #398

### CODE REVIEW

File: src/sidebars/favourites/tearout.js [hunk 1]
```diff
@@ -16,6 +16,7 @@
 
                         var windowService = window.windowService;
                         var tearoutWindow = windowService.createTearoutWindow(window.name);
+                        var me = {};
```
Comment (bjedrzejewski) [inline src/sidebars/favourites/tearout.js]: Why do you call this var `me`?

Comment (owennw) [inline src/sidebars/favourites/tearout.js]: That's mostly a hangover from the original tearout example, but not something that's changed in this PR.

Comment (bjedrzejewski) [inline src/sidebars/favourites/tearout.js]: Yeah I have seen it was already there, if you think it makes sense to give it better name, we can do it, otherwise I am VERY happy for you to merge it (as it fixes bug in my feature :D )

Comment (owennw) [inline src/sidebars/favourites/tearout.js]: Let's do that for another PR :)
